tests/cases/compiler/iterableWithNeverAsUnionMember.ts

```diff
@@ -0,0 +1,30 @@
+// @strict: true
+// @noEmit: true
+// @target: es5,esnext
```

Comment:
I think the `es5` target should produce similar errors but right now it doesnt produce any